### PR TITLE
fix(TDP-4083): add feedback during dictionary upload

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4083

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/components/home/home.html [hunk 1]
```diff
@@ -7,9 +7,6 @@
 <preparation-copy-move></preparation-copy-move>
 <preparation-creator></preparation-creator>
 <insertion-home></insertion-home>
-<step-progress 	modal-title="'ADD_NEW_DATASET' | translate"
-				ng-if="!$ctrl.state.home.preparations.creator.isVisible"
-				steps="$ctrl.state.dataset.uploadSteps"
->
+<step-progress ng-if="!$ctrl.state.home.preparations.creator.isVisible">
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/home/home.html]: you should put <ng-if="!$ctrl.state.home.preparations.creator.isVisible"> into the template to be more consistent with above components

File: dataprep-webapp/src/app/services/progress/progress-module.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const MODULE_NAME = 'data-prep.services.progress';
+
+/**
+ * @ngdoc object
+ * @name data-prep.services.progress
+ * @description This module contains the services in charge of progression
+ * @requires data-prep.services.utils
+ */
+angular.module(MODULE_NAME, [])
+	.service('ProgressConstants', ProgressConstants)
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/progress/progress-module.js]: I prefer to merge these two services into one service.

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/567/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/567/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/568/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/568/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/569/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/569/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/572/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/572/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/573/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/573/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/574/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/574/)

File: dataprep-webapp/src/app/services/state/dataset/dataset-state-service.js [hunk 3]
```diff
@@ -16,22 +16,11 @@ export const datasetState = {
 	uploadSteps: [],
 };
 
-export function DatasetStateService($translate) {
+export function DatasetStateService() {
 	'ngInject';
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/state/dataset/dataset-state-service.js]: useless

File: dataprep-webapp/src/app/services/state/progress/progress-state-service.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	 * @name start
+	 * @methodOf data-prep.services.state.service:ProgressStateService
+	 * @description Displays the step progress modal
+	 * @param {Object} schema The steps and the modal title
+	 * @param {Function} getter The getter to use to obtain the progress value
+	 */
+	function start(schema, getter) {
+		progressState.title = schema.title;
+		progressState.progressionGetter = getter || (() => 100);
+		progressState.steps = [...schema.steps].map((s) => {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/state/progress/progress-state-service.js]: avoid "s" , please

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/575/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/575/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/576/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/576/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/577/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/577/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/580/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/580/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/581/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/581/)

**Pull request merged**